tket/src/Circuit/basic_circ_manip.cpp

```diff
@@ -410,17 +437,14 @@ void Circuit::add_bit(const Bit& id, bool reject_dups) {
           "A unit with ID \"" + id.repr() + "\" already exists");
     } else if (found->type() == UnitType::Bit) {
       return;
-    } else {
-      throw CircuitInvalidity(
-          "A qubit with ID \"" + id.repr() + "\" already exists");
     }
   }
   opt_reg_info_t reg_info = get_reg_info(id.reg_name());
   register_info_t correct_info = {UnitType::Bit, id.reg_dim()};
-  if (reg_info && !(reg_info.value() == correct_info))
```

Comment:
This change was only made to reach the needed branch coverage:
The line was never reached in the tests and could only be reached by changing the code, I think it is the right thing to have an TKET_ASSERT in this case, replacing the throw exception.

Reply:
Seems this was the wrong decision: the code below can be reached!

Reply:
Yes, my assumption was based on the c++ coverage tests, looks like that was not the best indicator.